R/qualification-ddi.R

```diff
@@ -634,3 +634,33 @@ ddiPKRatioColumnName <- list(
   "CMAX" = "CmaxR Avg",
   "C_max" = "CmaxR Avg"
 )
+
+
+#' @title getGuestDeltaFromConfigurationPlan
+#' @description Get Guest et al. delta values for a DDI Ratio plot from the configuration plan
+#' @param ddiRatioPlan Properties defined in `DDIRatioPlots` field of confiuration plan
+#' @return A named list of Guest et al. values
+#' @keywords internal
+getGuestDeltaFromConfigurationPlan <- function(ddiRatioPlan) {
+  pkParameters <- ddiRatioPlan$PKParameters %||% ospsuite::toPathArray(ddiRatioPlan$PKParameter)
+  # By default all Guest et al.  delta values are 1
+  guestDelta <- as.list(sapply(pkParameters, function(x) {
+    1
+  }))
+  if (isEmpty(ddiRatioPlan$GuestDelta)) {
+    return(guestDelta)
+  }
+  # If defined as a numeric, all parameters get same delta
+  if (is.numeric(ddiRatioPlan$GuestDelta)) {
+    return(as.list(sapply(pkParameters, function(x) {
+      ddiRatioPlan$GuestDelta
+    })))
+  }
+  # Otherwise, update defined values
+  for (guestPlan in ddiRatioPlan$GuestDelta) {
+    for (pkParameter in guestPlan$PKParameters) {
+      guestDelta[[pkParameter]] <- guestPlan$Value
+    }
+  }
+  return(guestDelta)
```

Comment:
This should throw an error if `pkParameter `is not in the list `ddiRatioPlan$PKParameters `

Reply:
Maybe I'll find the way to catch this on the Qualification Plan Schema level, but I don't think it's possible there.